### PR TITLE
Remove bondedAmount filtering and fix skip logic for querying all delegators

### PR DESCRIPTION
**Pull request by yondonfu**
- Remove the `bondedAmount_not` where clause from the GraphQL query used to fetch all delegators from the subgraph. It is possible for a delegator to have bondedAmount = 0, but non-zero pending fees if the delegator fully unbonded, but did not withdraw its earned fees. The delegator should be able to claim those pending fees on L2 so it should be included in the snapshot
- Fix the skip logic for querying all delegators from the subgraph. Previously, the skip param in the GraphQL query was set to `delegators.length`. The problem was that `delegators.length` would be the length of the delegators array which only included delegators that already passed the filtering steps (i.e. EOA + orchestrator checks). So, a subsequent query with skip set to `delegators.length` could not only include delegators that were included in the response for the previous query. For example, if the first query returns 100 delegators and this is filtered down to 80 delegators then the second query would set skip to 80 - the first 20 delegators in the response to the second query would have also been included in the response for the first query
- Another related problem was the while loop condition `batchLength == 100`. Previously, `batchLength` was set to the length of the batch of delegators *after* filtering. So, you could end up with 100 delegators returned by the subgraph which should subsequent another query with the skip param updated, but if those 100 delegators are filtered down to 80 delegators, the while loop condition would evaluate to false and the second query would not be sent.

### CODE REVIEW

File: src/eth/rpc.ts [hunk 1]
```diff
@@ -50,7 +51,8 @@ const isEOA = async (address: string) => {
 }
 
 const isOrchestrator = (item) => {
-    return item.id === item.delegate.id;
+    // delegate could be null
+    return item.id === item.delegate?.id;
```
Comment (yondonfu) [inline src/eth/rpc.ts:55]: `delegate` will be null if the delegator fully unbonded such that it currently does not have a delegate.

**Review by kautukkundan — APPROVED** [review]
LGTM.

WDYT would be the best way to test edge cases?

**Comment by yondonfu** [comment]
> WDYT would be the best way to test edge cases?

One thing we could do is dump the list of addresses from the subgraph response and compare it with the list of addresses from a Dune Analytics query.
